include/SugarObjects/templates/basic/Basic.php

```diff
@@ -84,26 +85,71 @@ public function get_summary_text()
      */
     public function getEmailAddressConfirmOptInTick($emailField)
     {
-        $this->getEmailAddressValidateArguments($emailField);
+        global $sugar_config;
+        global $app_list_strings;
+        global $app_strings;
+        global $mod_strings;
 
-        global $sugar_config, $app_strings;
+        $this->getEmailAddressValidateArguments($emailField);
 
         $tickHtml = '';
-        $confirmOptIn = $this->getEmailAddressConfirmOptIn($emailField);
-        if ($confirmOptIn && $sugar_config['email_enable_confirm_opt_in']) {
-            $tickTitle = $app_strings['LBL_CONFIRM_OPT_IN_TITLE'];
-            $tickHtml = '<span class="confirm-opt-in-tick" title="' . $tickTitle . '">&#10004;</span>';
+
+        if ($sugar_config['email_enable_confirm_opt_in']) {
+            $template = new Sugar_Smarty();
+            $template->assign('APP', $app_strings);
+            $template->assign('APP_LIST_STRINGS', $app_list_strings);
+            $template->assign('MOD', $mod_strings);
+            $template->assign('OPT_IN_STATUS', $this->getEmailAddressOptInStatus($emailField));
+            $tickHtml = $template->fetch('include/SugarObjects/templates/basic/tpls/displayEmailAddressOptInField.tpl');
         }
 
         return $tickHtml;
     }
 
+
+    /**
+     * @global array $sugar_config
+     * @global \LoggerManager $log
+     * @param string $emailField
+     * @return string
+     */
+    public function getEmailAddressOptInStatus($emailField)
+    {
+        $emailAddress = $this->getEmailAddressConfirmOptIn($emailField);
+
+        $doNotDisplayOptInTickForModule = array(
```

Comment:
It seems like an external argument or parameter, separate it and/or make it configurable.

Reply:
I moved into a static variable which can be overridden if needed. I am not sure we want administrators changing this value.

Reply:
But the developers may want. Anyway, It's hard coded at the moment but it should be a function parameter.